source/agora/node/Ledger.d

```diff
@@ -105,19 +105,23 @@ public class Ledger
             }
         }
 
-        // restore validator set from lastest blocks
-        if (this.last_block.header.height > 0)
+        // +1 because the genesis block counts as one
+        const ulong block_count = this.last_block.header.height + 1;
+
+        // we are only interested in the last 1008 blocks,
+        // because that is the maximum length of an enrollment.
+        const ulong min_height =
+            block_count >= Enrollment.ValidatorCycle
+            ? block_count - Enrollment.ValidatorCycle : 0;
+
+        // restore validator set from the blockchain.
+        // using block_count, as the range is inclusive
```

Comment:
Well the range is not inclusive: the right bound is exclusive, but since you used `+ 1` it works.